### DISCUSSION TITLE
[datar][   INFO] `summarise()` has grouped output by ['sale'l] (override with `_groups` argument)

### DISCUSSION THREAD

**Discussion by antonio-yu** [pwwang/datar#124] [Q&A]
When will this warning message show?
When coding like this , the warning shows. 
`data >> groupy_by(f.col1) >> summarize(sum = f.col2.sum())`

**Comment by pwwang** [comment]
Here explains it:  https://dplyr.tidyverse.org/reference/summarise.html

> .groups
>   [Experimental] Grouping structure of the result.
>   - "drop_last": dropping the last level of grouping. This was the only supported option before version 1.0.0.
>   - "drop": All levels of grouping are dropped.
>   - "keep": Same grouping structure as .data.
>   - "rowwise": Each row is its own group.
>
> When .groups is not specified, it is chosen based on the number of rows of the results:
>   - If all the results have 1 row, you get "drop_last".
>   - If the number of rows varies, you get "keep".
> In addition, a message informs you of that choice, unless the result is ungrouped, the option "dplyr.summarise.inform" is set to FALSE, or when summarise() is called from a function in a package.

`.groups` corresponds to `_groups` argument in `datar`. You can use `options(dplyr_summarise_inform=False)` to always disable the message.

See also:

- https://stackoverflow.com/questions/62140483/how-to-interpret-dplyr-message-summarise-regrouping-output-by-x-override
- https://github.com/tidyverse/dplyr/issues/5852
